tests/test_iam/test_iam.py

```diff
@@ -756,5 +758,49 @@ def test_get_account_authorization_details():
     len(result['GroupDetailList']) == 1
     len(result['Policies']) > 1
 
+@mock_iam()
+def test_create_saml_provider():
+    conn = boto3.client('iam', region_name='us-east-1')
+    response = conn.create_saml_provider(
+        Name="TestSAMLProvider",
+        SAMLMetadataDocument='a' * 1024
+    )
+    response['SAMLProviderArn'].should.equal("arn:aws:iam::123456789012:saml-provider/TestSAMLProvider")
 
+@mock_iam()
+def test_get_saml_provider():
+    conn = boto3.client('iam', region_name='us-east-1')
+    saml_provider_create = conn.create_saml_provider(
+        Name="TestSAMLProvider",
+        SAMLMetadataDocument='a' * 1024
```

Comment:
This was done because you have to have a minimum of 1k characters in your SAML document. I had originally wanted to pass in a real SAML document, but then it started getting super crowded, or reading from the OS for a SAML file. Additionally, there appears to be some weird issues with having XML in XML when the botocore parser is invoked from Moto during a `make test` 🤷‍♀️ happy to dig further but not sure how much value we truly get from it.